amethyst_renderer/src/formats/texture.rs

```diff
@@ -176,36 +172,27 @@ impl PngFormat {
     }
 }
 
-impl Format<Texture> for PngFormat {
+impl SimpleFormat<Texture> for PngFormat {
     const NAME: &'static str = "JPEG";
 
     type Options = TextureMetadata;
 
-    fn import(
-        &self,
-        name: String,
-        source: Arc<Source>,
-        options: TextureMetadata,
-    ) -> Result<TextureData, BoxedErr> {
-        self.from_data(source.load(&name)?, options)
+    fn import(&self, bytes: Vec<u8>, options: TextureMetadata) -> Result<TextureData, BoxedErr> {
+        self.from_data(bytes, options)
     }
 }
 
 /// Allows loading of BMP files.
+#[derive(Clone)]
 pub struct BmpFormat;
 
-impl Format<Texture> for BmpFormat {
+impl SimpleFormat<Texture> for BmpFormat {
     const NAME: &'static str = "BMP";
 
     type Options = TextureMetadata;
 
-    fn import(
-        &self,
-        name: String,
-        source: Arc<Source>,
-        options: TextureMetadata,
-    ) -> Result<TextureData, BoxedErr> {
-        imagefmt::bmp::read(&mut Cursor::new(source.load(&name)?), ColFmt::RGBA)
+    fn import(&self, bytes: Vec<u8>, options: TextureMetadata) -> Result<TextureData, BoxedErr> {
+        imagefmt::bmp::read(&mut Cursor::new(bytes), ColFmt::RGBA)
```

Comment:
It's better to read BMP file directly into gpu visible memory.
But this can be addressed later. Just put a note, please.

Reply:
Will do.